### PR TITLE
Updated the kernel to the latest Symfony Flex recipe changes

### LINKED ISSUE #635: Unable to clear the cache due to fatal error
When I try to clear the cache with a Symfony console command:
```bash
bin/console cache:clear
```

I get an error:
> Fatal error:  Class App\Kerne_ may not inherit from final class (App\Kernel) in /Users/victor/www/sites/symfony-demo/var/cache/de_/kernel.tmp on line 49

Full output:
```text
bin/console cache:clear

 // Clearing the cache for the dev environment with debug true

PHP Fatal error:  Class App\Kerne_ may not inherit from final class (App\Kernel) in /Users/victor/www/sites/symfony-demo/var/cache/de_/kernel.tmp on line 49
PHP Stack trace:
PHP   1. {main}() /Users/victor/www/sites/symfony-demo/bin/console:0
PHP   2. Symfony\Bundle\FrameworkBundle\Console\Application->run() /Users/victor/www/sites/symfony-demo/bin/console:33
PHP   3. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() /Users/victor/www/sites/symfony-demo/vendor/symfony/console/Application.php:125
PHP   4. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() /Users/victor/www/sites/symfony-demo/vendor/symfony/framework-bundle/Console/Application.php:81
PHP   5. Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() /Users/victor/www/sites/symfony-demo/vendor/symfony/console/Application.php:224
PHP   6. Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->run() /Users/victor/www/sites/symfony-demo/vendor/symfony/console/Application.php:906
PHP   7. Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->execute() /Users/victor/www/sites/symfony-demo/vendor/symfony/console/Command/Command.php:264
PHP   8. Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->warmupCache() /Users/victor/www/sites/symfony-demo/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:85
PHP   9. Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->warmup() /Users/victor/www/sites/symfony-demo/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:121
PHP  10. Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->getTempKernel() /Users/victor/www/sites/symfony-demo/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:145

Fatal error: Class App\Kerne_ may not inherit from final class (App\Kernel) in /Users/victor/www/sites/symfony-demo/var/cache/de_/kernel.tmp on line 49

Call Stack:
    0.0002     360648   1. {main}() /Users/victor/www/sites/symfony-demo/bin/console:0
    0.0108    1599952   2. Symfony\Bundle\FrameworkBundle\Console\Application->run() /Users/victor/www/sites/symfony-demo/bin/console:33
    0.0245    1986416   3. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() /Users/victor/www/sites/symfony-demo/vendor/symfony/console/Application.php:125
    1.5266   26895952   4. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() /Users/victor/www/sites/symfony-demo/vendor/symfony/framework-bundle/Console/Application.php:81
    1.5267   26895952   5. Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() /Users/victor/www/sites/symfony-demo/vendor/symfony/console/Application.php:224
    1.5292   27066688   6. Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->run() /Users/victor/www/sites/symfony-demo/vendor/symfony/console/Application.php:906
    1.5294   27067704   7. Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->execute() /Users/victor/www/sites/symfony-demo/vendor/symfony/console/Command/Command.php:264
    1.5320   27178664   8. Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->warmupCache() /Users/victor/www/sites/symfony-demo/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:85
    1.5323   27178952   9. Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->warmup() /Users/victor/www/sites/symfony-demo/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:121
    1.5323   27179016  10. Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->getTempKernel() /Users/victor/www/sites/symfony-demo/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:145

18:30:05 CRITICAL  [php] Fatal Compile Error: Class App\Kerne_ may not inherit from final class (App\Kernel) ["exception" => Symfony\Component\Debug\Exception\FatalErrorException { …}] []


  [Symfony\Component\Debug\Exception\FatalErrorException]
  Compile Error: Class App\Kerne_ may not inherit from final class (App\Kernel)


cache:clear [--no-warmup] [--no-optional-warmers] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--] <command>
```

However, clearing the cache without warm up works well for me:
```bash
bin/console cache:clear --no-warmup
```

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #635 and updates the kernel according to https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/3.3/src/Kernel.php

### CODE REVIEW

File: src/Kernel.php [hunk 1]
```diff
@@ -55,7 +52,7 @@ public function getLogDir(): string
         return dirname(__DIR__).'/var/log';
     }
 
-    public function registerBundles(): iterable
+    public function registerBundles()
```
Comment (stof) [inline src/Kernel.php]: As the demo project requires PHP 7.1, you could keep this

Comment (javiereguiluz) [reply src/Kernel.php]: Yes, but I prefer to use the exact same code from the Symfony recipe. That's why we're also reverting the changes in our config (see #666).

Comment (sroze) [reply src/Kernel.php]: "PHP 5.5.9 or higher" is described in the `README.md` as a requirement.

Comment (stof) [reply src/Kernel.php]: well, the readme was forgotten when changing the requirement in the composer.json

Comment (javiereguiluz) [reply src/Kernel.php]: There's a pending PR for that: #625.

**Pull request merged**